ui/src/views/user/survey-edit/SurveyEditView.vue

```diff
@@ -240,22 +249,60 @@ export default {
       return this.formData;
     },
     onSaveAsDraftOptionClick() {
+      console.log("save as draft");
       const finalOutput = this.getData();
       console.log(JSON.stringify(finalOutput));
 
-      this.$notify.toast("Survey has been successfully saved");
+      if(finalOutput.config.startDate == "" || finalOutput.config.endDate == "") {
```

Comment:
I see that you added the validation here XD but we can also set the default value in the formData (my fault, I forgot). Just like BE, we can set the startDate as today and endDate as 7 days from now.

Reply:
this is the problematic part because BE currently has no boolean flag to indicate if survey is published or not, so for now i use startDate check to determine if survey is published

and because of this, we can't set a default value for startDate to today, because it will make the survey status immediately started/published (startDate < currentDate). what if the user only wants to save draft?

setting def value of startDate to some day in the future also won't make sense because it will make survey status appear published once the date is reached, and the user may not be aware of it

the only sensible solution imo is by asking the user to set the start end dates by themselves before saving

Reply:
..I dont know what to comment about that XD its really.. appealing in.. many senses rather than one. And not your fault, I understand your logic, but.. well...

Alright then.

Reply:
or maybe i can set so that the default startDate value on FE when saving as draft as year 2100. 🧶

Reply:
Please dont. :"

Reply:
I have finished refactoring my code, just give me a moment to test everything in docker and then I will push the changes. Thanks a bunch for your support!

Reply:
I have pushed the changes, please have a look. 😅 Thanks!

Reply:
I have checked and commented, please check my suggestions :"